daft/extension_type.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import annotations
+
+from typing import Callable
+
+from daft.dependencies import pa
+
+
+class DaftExtension(pa.ExtensionType):  # type: ignore[misc]
```

Comment:
I'd put an inline comment here saying something like "Never directly import this! Always use the `_ensure_registered_super_ext_type` function in `daft.datatype`."